src/main/java/co/rsk/bitcoinj/script/P2shErpFederationRedeemScriptParser.java

```diff
@@ -1,25 +1,58 @@
 package co.rsk.bitcoinj.script;
 
+import co.rsk.bitcoinj.core.BtcECKey;
+import co.rsk.bitcoinj.core.Sha256Hash;
 import co.rsk.bitcoinj.core.Utils;
 import co.rsk.bitcoinj.core.VerificationException;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.ArrayList;
 import java.util.List;
 
-public class P2shErpFederationRedeemScriptParser extends StandardRedeemScriptParser {
+public class P2shErpFederationRedeemScriptParser implements RedeemScriptParser {
     private static final Logger logger = LoggerFactory.getLogger(P2shErpFederationRedeemScriptParser.class);
 
     public static long MAX_CSV_VALUE = 65_535L; // 2^16 - 1, since bitcoin will interpret up to 16 bits as the CSV value
 
+    private final RedeemScriptParser standardRedeemScriptParser;
+
     public P2shErpFederationRedeemScriptParser(
         List<ScriptChunk> redeemScriptChunks
     ) {
-        super(
-            extractStandardRedeemScriptChunks(redeemScriptChunks)
-        );
-        this.multiSigType = MultiSigType.P2SH_ERP_FED;
+        List<ScriptChunk> standardRedeemScriptChunks = extractStandardRedeemScriptChunks(redeemScriptChunks);
+        this.standardRedeemScriptParser = new StandardRedeemScriptParser(standardRedeemScriptChunks);
+    }
+
+    @Override
+    public MultiSigType getMultiSigType() {
+        return MultiSigType.P2SH_ERP_FED;
+    }
+
+    @Override
+    public int getM() {
+        return standardRedeemScriptParser.getM();
+    }
+
+    @Override
+    public int findKeyInRedeem(BtcECKey key) {
+        return standardRedeemScriptParser.findKeyInRedeem(key);
+    }
+
+    @Override
+    public List<BtcECKey> getPubKeys() {
+        return standardRedeemScriptParser.getPubKeys();
+    }
+
+    @Override
+    public int findSigInRedeem(byte[] signatureBytes, Sha256Hash hash) {
+        return standardRedeemScriptParser.findSigInRedeem(signatureBytes, hash);
+    }
+
+    @Override
+    public List<ScriptChunk> extractStandardRedeemScriptChunks() {
+        return standardRedeemScriptParser.extractStandardRedeemScriptChunks();
     }
 
     public static List<ScriptChunk> extractStandardRedeemScriptChunks(List<ScriptChunk> chunks) {
```

Comment:
Very strange. Why is this method duplicated?

Reply:
Fixed.